grc/core/generator/cpp_top_block.py

```diff
@@ -28,31 +28,6 @@
 
 class CppTopBlockGenerator(TopBlockGenerator):
 
-    def __init__(self, flow_graph, file_path):
-        """
-        Initialize the C++ top block generator object.
-
-        Args:
-            flow_graph: the flow graph object
-            file_path: the path where we want to create
-                a new directory with C++ files
-        """
-
-        self._flow_graph = FlowGraphProxy(flow_graph)
-        self._generate_options = self._flow_graph.get_option('generate_options')
-
-        self._mode = TOP_BLOCK_FILE_MODE
-        # Handle the case where the directory is read-only
-        # In this case, use the system's temp directory
-        if not os.access(file_path, os.W_OK):
-            file_path = tempfile.gettempdir()
-
-        # When generating C++ code, we create a new directory
-        # (file_path) and generate the files inside that directory
-        filename = self._flow_graph.get_option('id')
```

Comment:
First generate python code of a flowgraph. Then try generate cpp code for the same flowgraph. This worked before, but fails now, as the cpp code generator tries to create a directory with the name of the generated python file.

Reply:
@haakov Hmm, it seems the original TopBlockGenerator was, I imagine due to the fact that there was no c++ output from grc, python specific. I wonder if it would be better to completely divorce the CppTopBlockGenerator from the TopBlockGenerator by setting its base to object. I can't find any dependencies on the TopBlockGenerator by inspecting the code at least.

Reply:
So there was one base method not overridden in CppTopBlockGenerator: _warnings. I added that to the class since it appears that TopBlockGenerator as written is python specific. Some refactoring may be useful to make TopBlockGenerator more generic, reduce duplicated code, and improve maintainability.